alphas/alpha_functions.py

```diff
@@ -29,3 +30,31 @@ def alpha_001(data: pd.DataFrame, ticker: str) -> pd.DataFrame:
     drop_signal_indices = data["actively_traded"].where(data["actively_traded"] == False).dropna().index
     raw_signal.loc[drop_signal_indices] = 0
     return raw_signal
+
+
+def alpha_042(data: pd.DataFrame, ticker: str) -> pd.DataFrame:
+    """
+    mult(
+        ite(
+            gt(mean_2(close),plus(mean_8(close),std_8(close))),
+            neg(const_1),
+            const_1),
+        csrank(div(volume,obv_20()))
+        )
+    Args: pd.DataFrame
+        Data: requires "Adj Close" and "Volume"
+    """
+    try:
+        return mult(
+            ite(
+                gt(
+                    mean_a(data, 2, column="Close")[6:],
```

Comment:
also, why is the 6 hardcoded? does this work for all cases? same for the case below. explain the hardcoded numbers

Reply:
They are hardcoded to make the dfs the same size. The [6:] is there since it compares a rolling transform with period 2 to a rolling transform of period 8. Since (after dropna), the first 6 values (dates) in the rolling transform with period 2 are not present in the one with period 8, so [6:] is to even out the list. Same thing with [13:] as it makes the lists the same size when multiplying a dataframe of rolling period 8 with a (ranked) dataframe of rolling period 20. 

They are hardcoded to match up the indices of the transformed data. They are replicable since the transforms are performed onto the original data, and they are chosen to avoid dimension/EOF errors.